### PR TITLE
[REACTOS] Properly check 1st NtQueryInformationToken() result

### PR DESCRIPTION
**Pull request by SergeGautherie**
(Everywhere) We don't want to continue with `size = 0`, do we?

Follow-up to #2857.

### CODE REVIEW

File: dll/win32/advapi32/reg/reg.c [hunk 1]
```diff
@@ -3492,7 +3492,7 @@ RegOpenUserClassesRoot(IN HANDLE hToken,
                                      NULL,
                                      0,
                                      &RequiredLength);
-    if (!NT_SUCCESS(Status) && (Status != STATUS_BUFFER_TOO_SMALL))
+    if (Status != STATUS_BUFFER_TOO_SMALL)
```
Comment (ThFabba) [inline dll/win32/advapi32/reg/reg.c:3495]: This is bogus. You're now returning success here but no valid key handle.

Comment (SergeGautherie) [reply dll/win32/advapi32/reg/reg.c:3495]: Agreed. Fixed.

File: dll/win32/advapi32/reg/reg.c [hunk 2]
```diff
@@ -3492,8 +3492,14 @@ RegOpenUserClassesRoot(IN HANDLE hToken,
...
     {
+        if (Status == STATUS_SUCCESS)
```
Comment (ThFabba) [inline dll/win32/advapi32/reg/reg.c:3497]: There are about 2 billion other success status codes.
I'd suggest
```
if (NT_SUCCESS(Status))
{
    ...
}
else if (Status != STATUS_BUFFER_TOO_SMALL)
{
    ...
}
```

Comment (SergeGautherie) [reply dll/win32/advapi32/reg/reg.c]: > `NT_SUCCESS(Status)`

Obviously. Fixed.

> `if else if`

While that is somewhat clearer to read, I feel it is better to have only one block of "error" handling, especially as we always expect `STATUS_BUFFER_TOO_SMALL` result.

**Comment by SergeGautherie** [comment]
> requested changes

@ThFabba, please update status.

**Pull request closed without merging**